openraft/src/core/raft_core.rs

```diff
@@ -369,7 +369,7 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> RaftCore<C,
 
             let my_id = self.id;
             let target_node = self.engine.state.membership_state.effective.get_node(&target).cloned();
-            let mut network = self.network.connect(target, target_node.as_ref()).await;
+            let mut network = self.network.connect(target, target_node.as_ref()).await.unwrap();
```

Comment:
The goal of returning a network error when calling `connect()` is not to panic an application. 

Thus here it should just discard this RPC and, instead of just a panic.